CARDS-2131: Increase test coverage cards-permissions

sofi2002sofi opened the pull request.

sofi2002sofi committed:
CARDS-2131: Increase test coverage cards-permissions

veronikaslc commented on modules/permissions/src/test/java/io/uhndata/cards/permissions/internal/QuestionnaireRestrictionPatternTest.java:
```diff
@@ -0,0 +1,131 @@
+    @Test
+    public void matchesForTreeWithoutResourceTypePropertyReturnsFalse()
+    {
+        NodeBuilderTree tree = new NodeBuilderTree(UUID.randomUUID().toString(), EmptyNodeState.EMPTY_NODE.builder());
```
missed test case of `tree` created from `sling:resourceType` other than `cards/Form`

veronikaslc commented on modules/permissions/src/test/java/io/uhndata/cards/permissions/internal/QuestionRestrictionPatternTest.java:
```diff
@@ -0,0 +1,129 @@
+    @Test
+    public void matchesForTreeWithoutResourceSuperTypePropertyReturnsFalse()
+    {
+        NodeBuilderTree tree = new NodeBuilderTree(UUID.randomUUID().toString(), EmptyNodeState.EMPTY_NODE.builder());
```
missed test case of `tree` created from `sling:resourceType` other than `cards/Answer`

veronikaslc commented on modules/permissions/src/test/java/io/uhndata/cards/permissions/internal/SectionRestrictionPatternTest.java:
```diff
@@ -0,0 +1,131 @@
+    @Test
+    public void matchesForTreeWithoutResourceSuperTypePropertyReturnsFalse()
+    {
+        NodeBuilderTree tree = new NodeBuilderTree(UUID.randomUUID().toString(), EmptyNodeState.EMPTY_NODE.builder());
```
missed test case of `tree` created from `sling:resourceType` other than `cards/AnswerSection`

veronikaslc commented:
`PermissionsManagerServiceTest` - > not covered cases from line 175 of `PermissionsManagerService`:

  `if (!restrictions.containsKey(key) || !entry.getRestriction(key).equals(restrictions.get(key)))`

veronikaslc commented:
`SessionSubjectRestrictionPatternTest` -> 2 of 4 branches missed for line 99 of `SessionSubjectRestrictionPattern`.

veronikaslc commented:
[jacoco.zip](https://github.com/data-team-uhn/cards/files/11775792/jacoco.zip)
`jacoco:report` html file attached

sofi2002sofi committed:
added new test cases for QuestionRestrictionPattern, QuestionnaireRestrictionPattern, SectionRestrictionPattern and SessionSubjectRestrictionPattern

sdumitriu committed:
[misc] Fix codestyle

sdumitriu committed:
Fix compiler warning

sofi2002sofi committed:
CARDS-2131: Increase test coverage of cards-permissions

sdumitriu committed:
CARDS-2131: Increase test coverage of cards-permissions
Modernize the revived tests and complete the module coverage to 100%:
- migrate the tests to the current platform APIs: the Jakarta servlet
  request built through the public Builders API, Mockito 5, sling-mock 4
- drop the Mockito runner in favor of plain mocks with reflection
  injection, since concurrently running runners collide on Mockito's
  global listener registry under the parallel surefire configuration
- remove the redundant osgi.cmpn dependency: it bundles an old copy of
  org.osgi.util.converter that shadows the one needed by the scripting
  BindingsValuesProvider tracker, breaking every Oak-backed test context
- raise the sling-mock resource resolver factory timeout, the default
  500ms is not always enough when several Oak-backed contexts start in
  parallel
- cover the previously untested paths: non-matching restriction targets,
  restriction comparison when removing an entry with restrictions,
  mismatched rule/principal/restriction removal attempts, forms without
  a subject, subjects with no matching ancestor, missing thread resource
  resolvers in the restriction factories
- drop the meaningless constructor smoke tests and rename the
  remaining vague test names
- remove the module's coverage exemption, restoring the default 1.00
  required instruction coverage ratio
